src/pages/user-projects/user-projects.jsx

```diff
@@ -0,0 +1,115 @@
+var React = require('react');
+var api = require('../../lib/api');
+var reportError = require('../../lib/errors');
+var i18n = require('../../lib/i18n');
+var router = require('../../lib/router');
+var platform = require('../../lib/platform');
+
+var render = require('../../lib/render.jsx');
+var Card = require('../../components/card/card.jsx');
+var Loading = require('../../components/loading/loading.jsx');
+
+var lang = i18n.isSupportedLanguage(i18n.currentLanguage) ? i18n.currentLanguage : i18n.defaultLang;
+
+var UserProjects = React.createClass({
+  mixins: [router,require('react-intl').IntlMixin],
+  getInitialState: function () {
+    return {
+      projects: [],
+      loading: false,
+      pagesLoaded: 0,
+      allPagesLoaded: false
+    };
+  },
+  load: function () {
+    this.setState({loading: true});
+
+    api({
+      uri: `/users/${this.state.params.user}/projects`,
+      useCache: true
+    }, (err, body) => {
+      this.setState({loading: false});
+
+      if (err) {
+        reportError(this.getIntlMessage('error_discovery_get'), err);
+        return;
+      }
+
+      if (!body || !body.projects || !body.projects.length) {
+        if (this.state.pagesLoaded === 0) {
+          reportError(this.getIntlMessage('error_featured_404'));
+        } else {
+          this.setState({
+            allPagesLoaded: true
+          });
+        }
+
+        return;
+      }
+
+      function shuffleArray(array) {
+        for (var i = array.length - 1; i > 0; i--) {
+          var j = Math.floor(Math.random() * (i + 1));
+          var temp = array[i];
+          array[i] = array[j];
+          array[j] = temp;
+        }
+
+        return array;
+      }
+
+      this.setState({
+        projects: this.state.projects.concat(shuffleArray(body.projects)),
```

Comment:
Good point. This was definitely something that just got hauled along with the duplication of the component, and I didn't parse through the logic enough to remove it since it was working.